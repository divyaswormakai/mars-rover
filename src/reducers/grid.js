import { ROVER_CLEAR } from '../actions/actionTypes';

const defaultState = { x: 9, y: 9 };

const grid = (state = defaultState, action) => {
  switch (action.type) {
    // Clear and give the default state
    case ROVER_CLEAR:
      return defaultState;

    default:
      return state;
  }
};

export default grid;
