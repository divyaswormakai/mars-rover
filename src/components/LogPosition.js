import React from 'react';
import { connect } from 'react-redux';
import styled from 'styled-components';

import { getRover } from '../reducers';

const getLogText = position => {
  if (typeof position === 'string') {
    return position;
  }
  return `X${position.x}, Y${position.y}`;
};

const LogPosition = ({ className, rover: { current, direction, log } }) => {
  return (
    <div className={className}>
      {current && (
        <div>
          <span className="current">Current position:</span>
          <strong>
            X{current.x}, Y{current.y}
          </strong>
        </div>
      )}
      <p className="direction">
        Current direction:
        <br />
        <b>{direction.toUpperCase()}</b>
      </p>
      {!!log.length && (
        <>
          <h5 className="title">Last positions: </h5>
          <div className="positions-container">
            {log.map((position, i) => (
              <p
                key={i}
              >
                {getLogText(position)}
              </p>
            ))}
          </div>
        </>
      )}
    </div>
  );
};

const StyledLogPosition = styled(LogPosition)`
  display: inline-block;
  height: 376px;
  overflow: hidden;
  border: 1px solid black;
  padding: 10px;
  min-width: 225px;
  border-radius: 10px;

  .current {
    display: block;
    margin: 0 0 5px;
  }

  .direction {
    margin: 20px 0 18px;
  }

  .delete-button {
    float: right;
    margin-right: 30px;
  }

  .title {
    margin: 23px 0 0 0;
  }

  .positions-container {
    height: 225px;
    overflow: auto;
  }

  
`;

const mapStateToProps = state => ({
  rover: getRover(state),
});

export default connect(mapStateToProps)(StyledLogPosition);
