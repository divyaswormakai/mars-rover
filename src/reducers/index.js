import { combineReducers } from 'redux';

import grid from './grid';
import rover from './rover';

const reducers = {
  grid,
  rover,
};

export const getGrid = state => state.grid;
export const getRover = state => state.rover;

export default combineReducers(reducers);
