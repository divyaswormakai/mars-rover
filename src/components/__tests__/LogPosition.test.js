import 'jest-styled-components';

import { render } from '@testing-library/react';
import React from 'react';
import ReactDOM from 'react-dom';
import { Provider } from 'react-redux';

import { getRover } from '../../reducers';
import state from '../../store/__mocks__/mockState';
import store from '../../store/__mocks__/mockStore';
import LogPosition from '../LogPosition';

jest.mock('../../actions/roverActions');
jest.mock('../../reducers');

describe('LogPosition test suite', () => {
  const component = (
    <Provider store={store}>
      <LogPosition />
    </Provider>
  );

  const setup = () => render(component);

  beforeEach(() => {
    getRover.mockReturnValue(state.rover);
  });

  afterEach(jest.clearAllMocks);

  it('Renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(component, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('Should match snapshot', () => {
    const { container } = setup();
    expect(container).toMatchSnapshot();
  });
});
