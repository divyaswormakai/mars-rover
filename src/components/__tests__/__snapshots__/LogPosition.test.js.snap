// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LogPosition test suite Should match snapshot 1`] = `
.c0 {
  display: inline-block;
  height: 376px;
  overflow: hidden;
  border: 1px solid black;
  padding: 10px;
  min-width: 225px;
  border-radius: 10px;
}

.c0 .current {
  display: block;
  margin: 0 0 5px;
}

.c0 .direction {
  margin: 20px 0 18px;
}

.c0 .delete-button {
  float: right;
  margin-right: 30px;
}

.c0 .title {
  margin: 23px 0 0 0;
}

.c0 .positions-container {
  height: 225px;
  overflow: auto;
}

<div>
  <div
    class="c0"
  >
    <div>
      <span
        class="current"
      >
        Current position:
      </span>
      <strong>
        X
        2
        , Y
        2
      </strong>
    </div>
    <p
      class="direction"
    >
      Current direction:
      <br />
      <b>
        EAST
      </b>
    </p>
    <h5
      class="title"
    >
      Last positions: 
    </h5>
    <div
      class="positions-container"
    >
      <p>
        X0, Y1
      </p>
      <p>
        X0, Y0
      </p>
    </div>
  </div>
</div>
`;
